prov/efa/src/rxr/rxr_pkt_type.h

```diff
@@ -390,6 +394,32 @@ static inline struct rxr_atomrsp_hdr *rxr_get_atomrsp_hdr(void *pkt)
 	return (struct rxr_atomrsp_hdr *)pkt;
 }
 
+/* receipt packet headers */
+struct rxr_receipt_hdr {
+	uint8_t type;
+	uint8_t version;
+	uint16_t flags;
+	/* end of rxr_base_hdr */
+	uint32_t tx_id;
```

Comment:
This should be based on msg id not tx id. In an ideal world we would use msg id everywhere in our protocol, but we're not there yet. If we don't add msg id now then we have to make another packet type in the future.

Reply:
`tx_id` is needed to retrive the `tx_entry` on which the message is send using `ofi_bufpool_get_ibuf`. 

In the initialization of `tx_entry`, it is `tx_id` is set using the `ofi_buf_index()`

Reply:
Yes, what I'm saying is you need to put both tx_id and msg_id in this header.

Reply:
Copy that. Shall I just introduce `msg_id` field or I need to set the `msg_id` in the receipt header as `tx_entry->msg_id` as well.

Reply:
I'm not sure I understand your question. The receipt packet should have the msg_id that it is acknowledging..

Reply:
I am confused whether I should set `receipt_hdr->msg_id` as `tx_entry->msg_id`. Or just add a field `msg_id` for future use.

Reply:
Either is fine.